onnx2tf/utils/common_functions.py

```diff
@@ -685,44 +716,96 @@ def explicit_broadcast(
                 input=const_or_var_2,
                 axis=0,
             )
-    transpose_perm = [0] + [i+2 for i in range(const_or_var_1_rank-2)] + [1]
-    if isinstance(const_or_var_2, np.ndarray):
-        const_or_var_2: np.ndarray = const_or_var_2.transpose(transpose_perm)
-        # # Check output values by brute force only if the shape of const_or_var_2
-        # # is different between const_or_var_1 and const_or_var_2 in dimensions other than 1
-        # const_or_var_1_shape_not_none = [
-        #     i if not isinstance(i, str) else 1 for i in const_or_var_1_shape
-        # ]
-        # for var1_shape, var2_shape in zip(const_or_var_1_shape_not_none, const_or_var_2.shape):
-        #     if var2_shape != 1 and var1_shape != 1 and var1_shape != var2_shape:
-        #         dummy_data_onnx = np.random.random_sample(graph_node.inputs[0].shape)
-        #         dummy_data_onnx = dummy_data_onnx.astype(graph_node.inputs[0].dtype)
-        #         dummy_data_tf = dummy_data_onnx.copy()
-        #         dummy_data_tf = dummy_data_tf.reshape(const_or_var_1_shape).astype(graph_node.inputs[0].dtype)
-        #         simple_arithmetic_validity_check(
-        #             op_type=graph_node.op,
-        #             onnx_x=dummy_data_onnx,
-        #             onnx_y=graph_node.inputs[1].values,
-        #             tf_x=dummy_data_tf,
-        #             tf_y=const_or_var_2,
-        #         )
-        #         break
-
-    elif isinstance(const_or_var_2, tf.Tensor) \
-        or (
-            not isinstance(const_or_var_2, np.ndarray) \
-            and tf.keras.backend.is_keras_tensor(const_or_var_2)
-        ):
-        if graph_node_input_name2 is not None \
-            and tf_layers_dict is not None \
-            and graph_node_input_name2 in tf_layers_dict \
-            and tf_layers_dict[graph_node_input_name2]['optype'] == 'Input':
-            const_or_var_2: np.ndarray = tf.transpose(
-                a=const_or_var_2,
-                perm=transpose_perm
-            )
-    else:
+        graph_node_input_shape2 = [1] + graph_node_input_shape2
+
+    # Check location of channel dimension in onnx and tflite
+    const_or_var_dim_check_1 = np.where(np.array(const_or_var_1.shape)[:, None]
+                                        == np.unique(const_or_var_1.shape)[None, :])[1]
+    graph_node_input_dim_check1 = np.where(np.array(graph_node_input_shape1)[:, None]
+                                           == np.unique(graph_node_input_shape1)[None, :])[1]
+    const_or_var_dim_check_2 = np.where(np.array(const_or_var_2.shape)[:, None]
+                                        == np.unique(const_or_var_2.shape)[None, :])[1]
+    graph_node_input_dim_check2 = np.where(np.array(graph_node_input_shape2)[:, None]
+                                           == np.unique(graph_node_input_shape2)[None, :])[1]
```

Comment:
>Omitted consideration of the case where the dimension contains an undefined dimension (None or UNK). Since we often convert models that contain undefined dimensions to saved_model, we would like to see None considered if possible.

I removed those unnecessary part, it was my misunderstood. None or UNK shape broadcasting now works like before. digits.onnx is now can be converted shown below.

<img src="https://user-images.githubusercontent.com/34959032/203182842-c30cdce8-5716-481e-8640-0ead128393f5.png" width="100" height="600">